### DISCUSSION TITLE
Q. Dto를 어디에서 관리할 것인가?

### DISCUSSION THREAD

**Discussion by jurlring** [woowacourse-teams/2022-gong-seek#44] [Q&A]
- Dto 패키지를 만들어서 관리할 것인가?
- 각 모듈에서 관리할 것인가?
    - 각 모듈중 어떤 패키지에서 관리하게 할 것인가?

**Comment by Gyuchool** [comment]
dto는 각 모듈마다 관리하는게 좋다고 생각합니다! 여러 모듈이 dto를 공유하진 않으니 담당 모듈에서 dto를 가지는게 관리하기 편할거같아여

**Comment by jurlring** [comment]
아직까진 controller와 service, controller와 view에서 사용하는 dto밖에 없다고 생각이 됩니다.
지금 dto를 사용하는 목적은 client에게 데이터를 전송할 때 묶어서 보내고 받을 수 있다는 효율성 때문에 사용하는 것이라고 생각이 되고, service에서 dto를 반환하는 것은 controller를 깔끔하게 유지하기 위함이 큰 것 같아서 계속 dto를 사용하는 이유가 동일하다면 지금처럼 모듈별로 layer 패키지 내부에 있는 것이 좋다고 생각해요.
현재는 presentation 패키지에 dto를 모아놓았는데, controller에서 service를 의존하고 있고 service에서 dto를 생성해서 사용하는 것이 양방향인 느낌이 들어서 application dto로 옮기는 것도 좋다고 생각합니다.

**Comment by brorae** [comment]
각 모듈에서 DTO를 관리하는 것이 좋다고 생각합니다. 서비스에서 requestDto, responseDto 모두 서비스에서 반환하고 사용하고 있기 때문에, application 패키지로 이동하는 것도 좋은 방법인 것 같습니다.

**Comment by hanull** [comment]
https://github.com/woowacourse-teams/2022-gong-seek/discussions/17#discussion-4192484 같은 디스커션인 것 같네요.

현재 presentation 계층에서 dto를 모두 관리하고 있는데, request와 response 모두 application 계층에서 관리하고 싶어요.

이유는 application 계층에서 dto를 관리한다면, 의존 방향을 단방향으로 처리할 수 있어요. (controller -> service)
현재 상황에서는 application 계층에서 presentation에 있는 dto를 의존하고 있기 때문에  application, presentation 계층간에 양방향 의존을 하게 돼요.

application 계층에서 관리하게 되면, controller 에서 service를 의존하고 있는 의존 방향을 단방향 그대로 유지할 수 있을 것 같아요.


- 참고
   - http://tech.pick-git.com/%EB%B3%80%EA%B2%BD%EC%97%90-%EC%9C%A0%EC%97%B0%ED%95%9C-%EC%84%A4%EA%B3%84,-%EC%9A%B0%EB%A6%AC%EB%8A%94-2%EA%B0%80%EC%A7%80-%EC%A2%85%EB%A5%98%EC%9D%98-DTO%EB%A5%BC-%EC%93%B4%EB%8B%A4/
